src/azurite/mod.rs

```diff
@@ -0,0 +1,247 @@
+use std::{borrow::Cow, collections::BTreeMap};
+
+use testcontainers::{
+    core::{ContainerPort, WaitFor},
+    Image,
+};
+
+const NAME: &str = "mcr.microsoft.com/azure-storage/azurite";
+const TAG: &str = "3.34.0";
+
+/// Port that [`Azurite`] uses internally for blob storage.
+pub const BLOB_PORT: ContainerPort = ContainerPort::Tcp(10000);
+
+/// Port that [`Azurite`] uses internally for queue.
+pub const QUEUE_PORT: ContainerPort = ContainerPort::Tcp(10001);
+
+/// Port that [`Azurite`] uses internally for table.
+const TABLE_PORT: ContainerPort = ContainerPort::Tcp(10002);
+
+const AZURITE_ACCOUNTS: &str = "AZURITE_ACCOUNTS";
+
+/// Module to work with [`Azurite`] inside tests.
+///
+/// This module is based on the official [`Azurite docker image`].
+///
+/// # Example
+/// ```
+/// use testcontainers_modules::{
+///     azurite,
+///     azurite::{Azurite, BLOB_PORT},
+///     testcontainers::runners::SyncRunner,
+/// };
+///
+/// let azurite = Azurite::default().start().unwrap();
+/// let blob_port = azurite.get_host_port_ipv4(BLOB_PORT).unwrap();
+///
+/// // do something with the started azurite instance..
+/// ```
+///
+/// [`Azurite`]: https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite?toc=%2Fazure%2Fstorage%2Fblobs%2Ftoc.json&bc=%2Fazure%2Fstorage%2Fblobs%2Fbreadcrumb%2Ftoc.json&tabs=visual-studio%2Cblob-storage
+/// [`Azurite docker image`]: https://hub.docker.com/r/microsoft/azure-storage-azurite
+#[derive(Debug, Clone)]
+pub struct Azurite {
+    tag: String,
+    env_vars: BTreeMap<String, String>,
+    loose: bool,
+    skip_api_version_check: bool,
+    disable_telemetry: bool,
+}
+
+impl Default for Azurite {
+    fn default() -> Self {
+        Self {
+            tag: TAG.to_string(),
+            env_vars: BTreeMap::new(),
+            loose: false,
+            skip_api_version_check: false,
+            disable_telemetry: false,
+        }
+    }
+}
+impl Azurite {
+    /// Create a new azurite instance with the latest image
+    pub fn latest() -> Self {
+        Self {
+            tag: "latest".to_string(),
+            ..Self::default()
+        }
+    }
+
+    /// Overrides the image tag.
+    /// Check https://mcr.microsoft.com/v2/azure-storage/azurite/tags/list to see available tags.
+    pub fn with_tag(self, tag: String) -> Self {
+        Self { tag, ..self }
+    }
```

Comment:
We usually don't provide such method here as `testcontainers-rs` allows this customization on core level via `ImageExt` trait

Reply:
I will remove it then. thk for the input